Scripts/Sources/Automation/Shell.swift

```diff
@@ -0,0 +1,113 @@
+import Foundation
+
+class Shell {
+    static func executeAndPrint(_ command: String, timeout: Double = 10, allowFailure: Bool = false) throws {
+        print("$ \(command)")
+        let output = try executeShell(commandPath: "/bin/bash", arguments: ["-c", command], timeout: timeout, allowFailure: allowFailure) {
+            print($0, separator: "", terminator: "")
+        }
+        if let error = output?.error {
+            print(error)
+        }
+    }
+
+    static func execute(_ command: String, timeout: Double = 10, allowFailure: Bool = false) throws -> (output: String?, error: String?)? {
+        return try executeShell(commandPath: "/bin/bash", arguments: ["-c", command], timeout: timeout, allowFailure: allowFailure)
+    }
+
+    static func bash(commandName: String,
+                     arguments: [String] = [],
+                     timeout: Double = 10,
+                     allowFailure: Bool = false) throws -> (output: String?, error: String?)? {
+        guard let execution = try? executeShell(commandPath: "/bin/bash" ,
+                                                arguments: [ "-l", "-c", "/usr/bin/which \(commandName)" ],
+                                                timeout: 1),
+            var whichPathForCommand = execution?.output else { return nil }
+
+        whichPathForCommand = whichPathForCommand.trimmingCharacters(in: NSCharacterSet.whitespacesAndNewlines)
+        return try executeShell(commandPath: whichPathForCommand, arguments: arguments, timeout: timeout, allowFailure: allowFailure)
+    }
+
+    static func executeShell(commandPath: String,
+                             arguments: [String] = [],
+                             timeout: Double = 10,
+                             allowFailure: Bool = false,
+                             stream: @escaping (String) -> Void = { _ in }) throws -> (output: String?, error: String?)? {
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/tevelee/Eval/blob/7d312c8f14aff0bcb5efb221edea46e4e917d2e9/Scripts/Sources/Automation/Shell.swift#L35"></span>Line should be 120 characters or less: currently 122 characters</td>
    </tr>
  </tbody>
</table>